hibernate-core/src/main/java/org/hibernate/mapping/Table.java

```diff
@@ -849,6 +849,9 @@ public int hashCode() {
 		}
 
 		public boolean equals(Object other) {
+			if (!(other instanceof ForeignKeyKey)) {
```

Comment:
This change here is fine, but please ensure you apply our code style: https://hibernate.org/community/contribute/intellij-idea/

Reply:
applied these rules